pyfar/dsp/filter/fractional_octaves.py

```diff
@@ -509,3 +529,595 @@ def reconstructing_fractional_octave_bands(
+        values of ``'order'``, the summed energy of the filter bank equals the
+        energy of input signal, i.e., the filter bank is energy preserving
+        (reconstructing). This is useful for analysis energetic properties of
+        the input signal such as the room acoustic property reverberation
+        time. For an amplitude preserving filter bank with -6 dB cut-off
+        frequencies see
+        :py:func:`~pyfar.dsp.filter.ReconstructingFractionalOctaveBands`.
+
+    Parameters
+    ----------
+    num_fractions : int, optional
+        The number of bands an octave is divided into. Eg., ``1`` refers to
+        octave bands and ``3`` to third octave bands. The default is ``1``.
+    frequency_range : array, tuple, optional
+        The lower and upper frequency limits. The default is
+        ``frequency_range=(20, 20e3)``.
+    order : int, optional
+        Order of the Butterworth filter. The default is ``14``.
+    sampling_rate : int, optional
+        The sampling rate in Hz. The default is ``44100``.
+
+    Examples
+    --------
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> # generate the data
+        >>> x = (pf.signals.impulse(2**17))
+        >>> FOFB = pf.dsp.filter.FractionalOctaveBands(num_fractions=1,
+        ...     frequency_range=(20, 8e3))
+        >>> y = FOFB.process(x)
+        >>> # frequency domain plot
+        >>> y_sum = pf.FrequencyData(np.sum(np.abs(y.freq)**2, 0),
+        ...     y.frequencies)
+        >>> pf.plot.freq(y)
+        >>> ax = pf.plot.freq(y_sum, color='k', log_prefix=10, linestyle='--')
+        >>> ax.set_title(
+        ...     "Filter bands and the sum of their squared magnitudes")
+        >>> plt.tight_layout()
+        >>> plt.show()
+
+    References
+    ----------
+    .. [#] International Electrotechnical Commission, International
+           Electrotechnical Commission und Technical Committee 29,
+           'Electroacoustics: Octave-Band and Fractional-Octave-Band
+           Filters.(IEC61260)', Part 1, Part 1, (2014),
+           ISBN: 978-2-8322-1417-6
+    """
+
+    def __init__(
+            self,
+            num_fractions=1,
+            frequency_range=(20.0, 20e3),
+            order=14,
+            sampling_rate=44100):
+
+        self._num_fractions = num_fractions
+        self._frequency_range = frequency_range
+        self._order = order
+        self._sampling_rate = sampling_rate
+
+        self._nominal_frequencies,\
+            self._center_frequencies,\
+            self._cutoff_frequencies = fractional_octave_frequencies(
+                self._num_fractions, self._frequency_range, True)
+
+        coefficients = self._get_coefficients(
+                sampling_rate=self._sampling_rate,
+                num_fractions=self._num_fractions,
+                frequency_range=self._frequency_range,
+                order=self._order)
```

Comment:
These do not necessarily need to be saved bit can be calculated in their respective getters depending on the parameters of the filter bank above.